### ISSUE TITLE
Error when calling CastMedia

### ISSUE THREAD

**Issue by neoassyrian** [react-native-google-cast/react-native-google-cast#185]
Hi guys, 

I'm getting an error message when calling CastMedia on ios. I'm not sure why. The state before the castMedia call is "connected" and the cast button shows that is connected to the device.

Error: No castSession!
  at castMedia(app:///main.jsbundle:735:488)

Any idea where the "No castSession!" is coming from? 

Keep in mind the same logic works fine on Android. I have also added "Access WiFi Information" capability in iOS.

**Comment by cgranetgithub** [comment]
I have the exact same problem.

**Comment by neoassyrian** [comment]
Are you streaming video, audio or live (e.g. Radio Stream)?
It started working for me. The only change I did was to add "isLive: true" because I was streaming a radio stream. 
Still don't know where that error was coming from

**Comment by cgranetgithub** [comment]
Thx for your feedback. I'm streaming video (HLS) hosted on CloudFront, so it is not live (actually it is VOD).
I tried with isLive, just in case, still the same error message...

**Comment by marcusstenbeck** [comment]
@neoassyrian @cgranetgithub 

tl;dr; Don't use `GoogleCast.getCastState()` to determine if a session is ready for `castMedia()`. Rely on the `react-native-google-cast` events instead.

Using `GoogleCast.getCastState()` doesn't seem to be a reliable way of knowing that there's a cast session.

The docs for [`react-native-google-cast`](https://github.com/react-native-google-cast/react-native-google-cast) never mention this as a way to determine connection status, and internally the code forwards the call to Google's library.

What I can tell from the iOS code in [RNGoogleCast.m](https://github.com/react-native-google-cast/react-native-google-cast/blob/master/ios/RNGoogleCast/RNGoogleCast.m) it lets you call that method before it has initialized its internal `castSession` variable.

**Comment by cgranetgithub** [comment]
@marcusstenbeck I tried to add those lines but it did not solve my problem.

However I managed to get rid of this "no cast session" issue by calling the "this.registerListeners()" method in the "componentDidMount()" of my component, as it is done in the example.
I suppose that it somehow establish the session...

**Comment by dcalhoun** [comment]
I believe I may have identified why some people experience this issue. I have outlined the details in https://github.com/react-native-google-cast/react-native-google-cast/issues/151#issuecomment-653056964.

**Comment by demianp** [comment]
I fixed a similar issue. You can check here:
https://github.com/react-native-google-cast/react-native-google-cast/pull/188#issuecomment-682090063

**Comment by petrbela** [comment]
v4 should be syncing the current state more correctly so I think this issue will go away. If you still experience it after updating, please create a new issue with up-to-date description.
